runtime/src/bucket_map_holder.rs

```diff
@@ -124,22 +124,35 @@ impl<T: IndexValue> BucketMapHolder<T> {
     }
 
     pub fn bucket_flushed_at_current_age(&self) {
-        self.count_buckets_flushed.fetch_add(1, Ordering::Release);
-        self.maybe_advance_age();
+        let count_buckets_flushed = 1 + self.count_buckets_flushed.fetch_add(1, Ordering::Release);
```

Comment:
Since now we're using the atomic's loaded value, can the ordering be `AcqRel` instead?